core/controllers/topics_and_skills_dashboard_test.py

```diff
@@ -174,13 +174,105 @@ def test_post(self):
 
         json_response = self.post_json(
             feconf.SKILL_DASHBOARD_DATA_URL, {
-                'items_per_page': '10',
+                'num_items_to_fetch': '10',
                 'sort': 'Oldest Created'
             }, csrf_token=csrf_token)
 
         self.assertEqual(len(json_response['skill_summary_dicts']), 2)
         self.assertEqual(
             json_response['skill_summary_dicts'][0]['id'], self.linked_skill_id)
+        self.assertEqual(
+            json_response['skill_summary_dicts'][1]['id'],
+            self.subtopic_skill_id)
+        self.assertFalse(json_response['more'])
+        self.assertEqual(json_response['next_cursor'], None)
+
+        json_response = self.post_json(
+            feconf.SKILL_DASHBOARD_DATA_URL, {
+                'num_items_to_fetch': '10',
+                'sort': 'Newly Created'
+            }, csrf_token=csrf_token)
+
+        self.assertEqual(len(json_response['skill_summary_dicts']), 2)
+        self.assertEqual(
+            json_response['skill_summary_dicts'][0]['id'],
+            self.subtopic_skill_id)
+        self.assertEqual(
+            json_response['skill_summary_dicts'][1]['id'], self.linked_skill_id)
+
+        json_response = self.post_json(
+            feconf.SKILL_DASHBOARD_DATA_URL, {
+                'num_items_to_fetch': '10',
+                'sort': 'Most Recently Updated'
+            }, csrf_token=csrf_token)
+
+        self.assertEqual(len(json_response['skill_summary_dicts']), 2)
+        self.assertEqual(
+            json_response['skill_summary_dicts'][0]['id'],
+            self.subtopic_skill_id)
+        self.assertEqual(
+            json_response['skill_summary_dicts'][1]['id'], self.linked_skill_id)
+        self.assertFalse(json_response['more'])
+        self.assertEqual(json_response['next_cursor'], None)
+
+    def test_post_with_keywords(self):
```

Comment:
The names of these tests are not descriptive enough. Try to add some info about what you are expecting the method to do / what is this test for.

I also don't see tests with more = True or a non-empty next cursor. Please add them for completeness.

I also don't see tests for the error cases (invalid args, missing args, etc.)

As I mentioned before, your tests need to be more comprehensive please.

Reply:
Improved them. Done(Hopefully)